installer/cluster.go

```diff
@@ -487,14 +631,49 @@ func (c *BaseCluster) bootstrapTarget(t *TargetServer) error {
 			break
 		}
 	}
-	if keyData.Key == "" || controllerCertData.Pin == "" {
+
+	if err := sess.Wait(); err != nil {
 		return err
 	}
 
-	c.ControllerKey = keyData.Key
-	c.ControllerPin = controllerCertData.Pin
-	c.CACert = controllerCertData.CACert
-	c.DashboardLoginToken = loginTokenData.Token
+	if c.backupPath != "" {
+		routerEnv, err := c.getAppEnv("router", t)
+		if err != nil {
+			return err
+		}
+		dashboardEnv, err := c.getAppEnv("dashboard", t)
+		if err != nil {
+			return err
+		}
+
+		b, _ := pem.Decode([]byte(routerEnv["TLSCERT"]))
+		if b == nil {
+			return fmt.Errorf("Error parsing TLSCERT")
+		}
+		sha := sha256.Sum256(b.Bytes)
+		c.ControllerPin = base64.StdEncoding.EncodeToString(sha[:])
```

Comment:
This isn't being used before the domain migration anymore, so no need to fetch the router env or calculate the pin.